### DISCUSSION TITLE
nohup: failed to run command

### DISCUSSION THREAD

**Discussion by whitequill** [pulsar-edit/.github#162] [Pulsar Support]
I'm getting the following error, `nohup: failed to run command '/tmp/pulsar-build/Pulsar/pulsar': No such file or directory` when I run `pulsar.sh`.
I am unsure of how to proceed. I'm running Gentoo.

**Comment by Spiker985** [comment]
It didn't find your pulsar file within `/opt/Pulsar/` and fell back to the original Atom build script directory

**Reply by whitequill** [reply]
I copied the files to /opt/Pulsar from the git pull. It gives the same error. I have never had atom installed on this system.

**Reply by Spiker985** [reply]
So, the development build sequence (git clone, yarn install, yarn build, yarn build:apm) doesn't currently work with pulsar.sh

I have rewritten the sh a few times, but getting it to properly pass everything to electron is somewhat of a nuisance

Use `yarn start` in order to launch the editor from the dev sequence

**Comment by Daeraxa** [comment]
Why are you trying to run `pulsar.sh`?

**Reply by whitequill** [reply]
I am unused to running/building Electron applications and `pulsar.sh` is the only executable in the directory.
I'm used to build systems like make and meson/ninja.  It's the first thing I think of when I find some new code. I'm simply unfamiliar with the ecosystem of Electron apps at this time.

**Reply by confused-Techie** [reply]
@whitequill no worries.
What I'd recommend then to get a little bit familiar, is to take a look at our [Downloads](https://pulsar-edit.dev/download.html) page, and if you'd like take a look at [Building Pulsar](https://pulsar-edit.dev/docs/launch-manual/sections/core-hacking/#building-pulsar) in our docs.

Essentially, we distribute easily installable versions of the application. Which you should be able to install like any `.exe`, `.deb`, `.appimage` and so on.

Alternatively, you can build the application from source (like what you are attempting to do now), which is used to develop the application, or could be used by some user if they prefer.

**Reply by Daeraxa** [reply]
@whitequill ok that makes more sense. But yeah, check @confused-Techie's links above for the build instructions. In a more general sense you can often find how to build or run electron (and node apps in general) by the scripts section inside the package.json file. For example with pulsar we have `yarn start` for starting the project and `yarn dist` for building binaries but for Pulsar check out the docs as there are a few extra bits you need to do.

**Reply by whitequill** [reply]
@confused-Techie Thank you for the links. I have been able to successfully build pulsar now.
The first place I look for build instructions is in the read-me or other files in a repository.  Would it be odd to ask that a markdown file with the building instructions be added to the repository?

**Reply by confused-Techie** [reply]
@whitequill So we do have `./docs/Building.md` which while not the most detailed, do include some of the correct steps to build the repo.

Although I'll be honest we are putting a lot of effort into getting the website docs updated, and I have a feeling we'd like to only have to worry about one place for them, the next best thing besides the docs there already is, could be to at least make those docs more obvious in the readme, and likely take more opportunities to link to the website which should always have more, and more thorough documentation

**Reply by whitequill** [reply]
> @whitequill So we do have `./docs/Building.md` which while not the most detailed, do include some of the correct steps to build the repo.
Thanks. I didn't think to look there for some reason.
